test/TestUserAcceptance.hs

```diff
@@ -37,20 +38,24 @@ data Condition
 systemTests ::
   TestTree
 systemTests =
-    testGroup
-        "user acceptance tests"
-        [ testUserViewsMailSuccessfully
-        , testUserCanManipulateNMQuery
-        , testUserCanSwitchBackToIndex
-        , testCanToggleHeaders
-        , testSetsMailToRead
-        , testErrorHandling
-        , testHelp
-        , testManageTagsOnMails
-        , testManageTagsOnThreads
-        ]
-
-testManageTagsOnMails :: TestTree
+  withResource pre post $ \_ ->
```

Comment:
heh ... funny how "easy" that is to just add group specific setup and teardown functions ...

Reply:
Yeah I was pleased with ease of discovery and ease of use of the interface.